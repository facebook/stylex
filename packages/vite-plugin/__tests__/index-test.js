/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 *
 */

'use strict';

const path = require('path');
const stylexPlugin = require('../lib/index.js');

describe('vite-stylex-plugin', () => {
  async function runStylex(options = {}) {
    const { build } = await import('vite');

    console.log(stylexPlugin);
    const bundle = await build({
      build: {
        lib: {
          entry: path.resolve(__dirname, '__fixtures__/index.js'),
          formats: ['es'],
          fileName: 'bundle',
        },
        cssMinify: false,
        minify: false,
        rollupOptions: {
          external: ['stylex'],
        },
        outDir: '__builds__',
        write: false,
      },
      plugins: [stylexPlugin(options)],
      logLevel: 'silent',
    });
    // Generate output specific code in-memory
    // You can call this function multiple times on the same bundle object
    let css, js;
    const { output } = bundle[0];
    for (const chunkOrAsset of output) {
      if (chunkOrAsset.fileName.includes('.css')) {
        css += chunkOrAsset.source;
      } else if (chunkOrAsset.fileName === 'bundle.mjs') {
        js = chunkOrAsset.code;
      }
    }
    return { css, js };
  }

  it('extracts CSS and removes stylex.inject calls', async () => {
    const { css, js } = await runStylex();
    expect(css).toMatchSnapshot();
    expect(js).toMatchSnapshot();
  });
});
