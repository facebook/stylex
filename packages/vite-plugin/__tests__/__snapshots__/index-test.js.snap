// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vite-stylex-plugin extracts CSS and removes stylex.inject calls 1`] = `
"undefined@keyframes xgnty7z-B{0%{opacity:.25;}100%{opacity:1;}}
.x1oz5o6v:hover:not(#\\#):not(#\\#){background:red}
.xeuoslp:not(#\\#):not(#\\#):not(#\\#){animation-name:xgnty7z-B}
.x1lliihq:not(#\\#):not(#\\#):not(#\\#){display:block}
.x78zum5:not(#\\#):not(#\\#):not(#\\#){display:flex}
.xt0psk2:not(#\\#):not(#\\#):not(#\\#){display:inline}
.x1hm9lzh:not(#\\#):not(#\\#):not(#\\#){margin-inline-start:10px}
.x1egiwwb:not(#\\#):not(#\\#):not(#\\#):not(#\\#){height:500px}
.xlrshdv:not(#\\#):not(#\\#):not(#\\#):not(#\\#){margin-top:99px}
.xh8yej3:not(#\\#):not(#\\#):not(#\\#):not(#\\#){width:100%}
.x3hqpx7:not(#\\#):not(#\\#):not(#\\#):not(#\\#){width:50%}"
`;

exports[`vite-stylex-plugin extracts CSS and removes stylex.inject calls 2`] = `
"import stylex from "stylex";
const styles$2 = {
  bar: {
    display: "x1lliihq",
    width: "xh8yej3",
    $$css: true
  }
};
const styles$1 = {
  baz: {
    display: "xt0psk2",
    height: "x1egiwwb",
    width: "x3hqpx7",
    $$css: true
  }
};
const styles = {
  foo: {
    animationName: "xeuoslp",
    display: "x78zum5",
    marginInlineStart: "x1hm9lzh",
    marginLeft: null,
    marginRight: null,
    marginTop: "xlrshdv",
    height: "x1egiwwb",
    ":hover_background": "x1oz5o6v",
    ":hover_backgroundAttachment": null,
    ":hover_backgroundClip": null,
    ":hover_backgroundColor": null,
    ":hover_backgroundImage": null,
    ":hover_backgroundOrigin": null,
    ":hover_backgroundPosition": null,
    ":hover_backgroundPositionX": null,
    ":hover_backgroundPositionY": null,
    ":hover_backgroundRepeat": null,
    ":hover_backgroundSize": null,
    $$css: true
  }
};
function App() {
  return stylex(styles$2.bar, styles.foo, styles$1.baz);
}
export {
  App as default
};
"
`;
